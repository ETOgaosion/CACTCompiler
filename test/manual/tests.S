.text
.align 1
.globl	main
.type	main, @function
main:
L_1_main:
	lw	t0, -20(sp)
	addi	t1, zero, 10
	slt	t2, t0, t1
	sb	t2, -28(sp)
	lw	t0, -16(sp)
	addi	t1, zero, 1
	slt	t2, t0, t1
	sb	t2, -36(sp)
	lb	t0, -28(sp)
	lb	t1, -36(sp)
	and	t2, t0, t1
	sb	t2, -40(sp)
	lb	t0, -40(sp)
	beqz	t0, L_0_main
	lw	t0, -20(sp)
	addi	t1, zero, 5
	slt	t2, t0, t1
	sb	t2, -44(sp)
	lb	t0, -44(sp)
	beqz	t0, L_2_main
	lw	t1, -20(sp)
	addi	t0, t1, 2
	sw	t0, -48(sp)
	lw	t0, -48(sp)
	sw	t0, -16(sp)
	lw	a0, -20(sp)
	lw	a0, -16(sp)
	addi	sp, sp, -56
	call	add_int
	addi	sp, sp, 56
	sw	a0, -52(sp)
	lla	t0, L_3_main
	jr	t0
L_2_main:
L_3_main:
	lw	t1, -20(sp)
	addi	t0, t1, 2
	sw	t0, -32(sp)
	lw	t0, -32(sp)
	sw	t0, -20(sp)
	lw	t1, -16(sp)
	addi	t0, t1, 1
	sw	t0, -24(sp)
	lw	t0, -24(sp)
	sw	t0, -16(sp)
	lla	t0, L_1_main
	jr	t0
L_0_main:
	mv	a0, zero
	ret
.text
.align 1
.globl	add_int
.type	add_int, @function
add_int:
	sw	a0, -16(sp)
	sw	a0, -12(sp)
	lw	t1, -16(sp)
	lw	t2, -12(sp)
	addw	t0, t1, t2
	sw	t0, -24(sp)
	lw	t0, -24(sp)
	sw	t0, -20(sp)
	lw	a0, -20(sp)
	ret
