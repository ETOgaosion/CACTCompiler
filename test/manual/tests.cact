int a = 2;
float m[4] = {1.0, 2.0, 3.0, 4.0};
int fun(int m) {
    return m + 1;
}
int main() {
    a = 5;
    int b = 0;
    while (a < 6 || b == 0) {
        if (a + b >= 10 && ((a * b) % 10) != 0) {
            break;
        }
        b = a * 2 + 10 + 12 + fun(5);
        a = a + 1;
    }
    print_int(b);
    return b;
}