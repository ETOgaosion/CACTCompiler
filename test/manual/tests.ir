============ global var ================
============ imm value ================
============ functions ================
----------- funcDef -----------
main() => 1:
----------- paramVars -----------
----------- localVars -----------
----------- tempVars ------------
t_0:= symbol type: TEMP VAR; data type: 1
----------- Codes ------------
main:
	addparam 1,1;
	addparam 2,2;
	addparam 3,3;
	addparam 4,4;
	addparam 5,5;
	addparam 6,6;
	addparam 7,7;
	call loop3; 
	t_0 =  return ;
	addparam t_0,1;
	call putint; 
	return 0;

----------- funcDef -----------
loop3(1,1,1,1,1,1,1) => 1:
----------- paramVars -----------
m := symbol type: 0; data type: 1
l := symbol type: 0; data type: 1
j := symbol type: 0; data type: 1
n := symbol type: 0; data type: 1
i := symbol type: 0; data type: 1
k := symbol type: 0; data type: 1
h := symbol type: 0; data type: 1
----------- localVars -----------
ans := symbol type: 1; data type: 1; initValue: 0

f := symbol type: 1; data type: 1; initValue: 0

e := symbol type: 1; data type: 1; initValue: 0

d := symbol type: 1; data type: 1; initValue: 0

c := symbol type: 1; data type: 1; initValue: 0

g := symbol type: 1; data type: 1; initValue: 0

b := symbol type: 1; data type: 1; initValue: 0

a := symbol type: 1; data type: 1; initValue: 0

----------- tempVars ------------
----------- Codes ------------
loop3:
	h =  param 1;
	i =  param 2;
	j =  param 3;
	k =  param 4;
	l =  param 5;
	m =  param 6;
	n =  param 7;
	a = 0;
	b = 0;
	c = 0;
	d = 0;
	e = 0;
	f = 0;
	g = 0;
	ans = 0;
	t_0 = 0;
	return t_0;

