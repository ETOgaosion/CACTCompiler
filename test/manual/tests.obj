
test/manual/tests:     file format elf64-littleriscv


Disassembly of section .text:

00000000000100b0 <register_fini>:
   100b0:	00000793          	li	a5,0
   100b4:	c789                	beqz	a5,100be <register_fini+0xe>
   100b6:	6541                	lui	a0,0x10
   100b8:	40e50513          	addi	a0,a0,1038 # 1040e <__libc_fini_array>
   100bc:	a659                	j	10442 <atexit>
   100be:	8082                	ret

00000000000100c0 <_start>:
   100c0:	00002197          	auipc	gp,0x2
   100c4:	c4818193          	addi	gp,gp,-952 # 11d08 <__global_pointer$>
   100c8:	f6018513          	addi	a0,gp,-160 # 11c68 <completed.1>
   100cc:	f9818613          	addi	a2,gp,-104 # 11ca0 <__BSS_END__>
   100d0:	8e09                	sub	a2,a2,a0
   100d2:	4581                	li	a1,0
   100d4:	1ce000ef          	jal	ra,102a2 <memset>
   100d8:	00000517          	auipc	a0,0x0
   100dc:	36a50513          	addi	a0,a0,874 # 10442 <atexit>
   100e0:	c519                	beqz	a0,100ee <_start+0x2e>
   100e2:	00000517          	auipc	a0,0x0
   100e6:	32c50513          	addi	a0,a0,812 # 1040e <__libc_fini_array>
   100ea:	358000ef          	jal	ra,10442 <atexit>
   100ee:	14a000ef          	jal	ra,10238 <__libc_init_array>
   100f2:	4502                	lw	a0,0(sp)
   100f4:	002c                	addi	a1,sp,8
   100f6:	4601                	li	a2,0
   100f8:	102000ef          	jal	ra,101fa <main>
   100fc:	aa39                	j	1021a <exit>

00000000000100fe <__do_global_dtors_aux>:
   100fe:	f601c703          	lbu	a4,-160(gp) # 11c68 <completed.1>
   10102:	e715                	bnez	a4,1012e <__do_global_dtors_aux+0x30>
   10104:	1141                	addi	sp,sp,-16
   10106:	e022                	sd	s0,0(sp)
   10108:	843e                	mv	s0,a5
   1010a:	e406                	sd	ra,8(sp)
   1010c:	00000793          	li	a5,0
   10110:	cb81                	beqz	a5,10120 <__do_global_dtors_aux+0x22>
   10112:	6545                	lui	a0,0x11
   10114:	4ec50513          	addi	a0,a0,1260 # 114ec <__FRAME_END__>
   10118:	00000097          	auipc	ra,0x0
   1011c:	000000e7          	jalr	zero # 0 <register_fini-0x100b0>
   10120:	4785                	li	a5,1
   10122:	60a2                	ld	ra,8(sp)
   10124:	f6f18023          	sb	a5,-160(gp) # 11c68 <completed.1>
   10128:	6402                	ld	s0,0(sp)
   1012a:	0141                	addi	sp,sp,16
   1012c:	8082                	ret
   1012e:	8082                	ret

0000000000010130 <frame_dummy>:
   10130:	00000793          	li	a5,0
   10134:	cb91                	beqz	a5,10148 <frame_dummy+0x18>
   10136:	6545                	lui	a0,0x11
   10138:	f6818593          	addi	a1,gp,-152 # 11c70 <object.0>
   1013c:	4ec50513          	addi	a0,a0,1260 # 114ec <__FRAME_END__>
   10140:	00000317          	auipc	t1,0x0
   10144:	00000067          	jr	zero # 0 <register_fini-0x100b0>
   10148:	8082                	ret

000000000001014a <iobuff>:
	...

000000000001019a <add>:
.text
.align 1
.globl	add
.type	add, @function
add:
	sd	ra, -8(sp)
   1019a:	fe113c23          	sd	ra,-8(sp)
	sw	a0, -12(sp)
   1019e:	fea12a23          	sw	a0,-12(sp)
	lw	t0, -12(sp)
   101a2:	ff412283          	lw	t0,-12(sp)
	addi	t1, zero, 2
   101a6:	4309                	li	t1,2
	xor	t2, t0, t1
   101a8:	0062c3b3          	xor	t2,t0,t1
	seqz	t3, t2
   101ac:	0013be13          	seqz	t3,t2
	sb	t3, -24(sp)
   101b0:	ffc10423          	sb	t3,-24(sp)
	lb	t0, -24(sp)
   101b4:	fe810283          	lb	t0,-24(sp)
	beqz	t0, L_0_add
   101b8:	02028363          	beqz	t0,101de <L_0_add>
	lw	t0, -12(sp)
   101bc:	ff412283          	lw	t0,-12(sp)
	addi	t1, zero, 2
   101c0:	4309                	li	t1,2
	mulw	t2, t0, t1
   101c2:	026283bb          	mulw	t2,t0,t1
	sw	t2, -20(sp)
   101c6:	fe712623          	sw	t2,-20(sp)
	lw	a0, -20(sp)
   101ca:	fec12503          	lw	a0,-20(sp)
	ld	ra, -8(sp)
   101ce:	ff813083          	ld	ra,-8(sp)
	ret
   101d2:	8082                	ret
	lla	t0, L_1_add
   101d4:	00000297          	auipc	t0,0x0
   101d8:	02028293          	addi	t0,t0,32 # 101f4 <L_1_add>
	jr	t0
   101dc:	8282                	jr	t0

00000000000101de <L_0_add>:
L_0_add:
	lw	t1, -12(sp)
   101de:	ff412303          	lw	t1,-12(sp)
	addi	t0, t1, 10
   101e2:	00a30293          	addi	t0,t1,10 # 1014a <iobuff>
	sw	t0, -16(sp)
   101e6:	fe512823          	sw	t0,-16(sp)
	lw	a0, -16(sp)
   101ea:	ff012503          	lw	a0,-16(sp)
	ld	ra, -8(sp)
   101ee:	ff813083          	ld	ra,-8(sp)
	ret
   101f2:	8082                	ret

00000000000101f4 <L_1_add>:
L_1_add:
	ld	ra, -8(sp)
   101f4:	ff813083          	ld	ra,-8(sp)
	ret
   101f8:	8082                	ret

00000000000101fa <main>:
.text
.align 1
.globl	main
.type	main, @function
main:
	sd	ra, -8(sp)
   101fa:	fe113c23          	sd	ra,-8(sp)
	mv	t0, zero
   101fe:	00000293          	li	t0,0
	beqz	t0, L_0_main
   10202:	00028763          	beqz	t0,10210 <L_0_main>
	lla	t0, L_1_main
   10206:	00000297          	auipc	t0,0x0
   1020a:	00a28293          	addi	t0,t0,10 # 10210 <L_0_main>
	jr	t0
   1020e:	8282                	jr	t0

0000000000010210 <L_0_main>:
L_0_main:
L_1_main:
	mv	a0, zero
   10210:	00000513          	li	a0,0
	ld	ra, -8(sp)
   10214:	ff813083          	ld	ra,-8(sp)
	ret
   10218:	8082                	ret

000000000001021a <exit>:
   1021a:	1141                	addi	sp,sp,-16
   1021c:	4581                	li	a1,0
   1021e:	e022                	sd	s0,0(sp)
   10220:	e406                	sd	ra,8(sp)
   10222:	842a                	mv	s0,a0
   10224:	128000ef          	jal	ra,1034c <__call_exitprocs>
   10228:	f481b503          	ld	a0,-184(gp) # 11c50 <_global_impure_ptr>
   1022c:	6d3c                	ld	a5,88(a0)
   1022e:	c391                	beqz	a5,10232 <exit+0x18>
   10230:	9782                	jalr	a5
   10232:	8522                	mv	a0,s0
   10234:	28e000ef          	jal	ra,104c2 <_exit>

0000000000010238 <__libc_init_array>:
   10238:	1101                	addi	sp,sp,-32
   1023a:	e822                	sd	s0,16(sp)
   1023c:	e04a                	sd	s2,0(sp)
   1023e:	6445                	lui	s0,0x11
   10240:	6945                	lui	s2,0x11
   10242:	4f040793          	addi	a5,s0,1264 # 114f0 <__init_array_start>
   10246:	4f090913          	addi	s2,s2,1264 # 114f0 <__init_array_start>
   1024a:	40f90933          	sub	s2,s2,a5
   1024e:	ec06                	sd	ra,24(sp)
   10250:	e426                	sd	s1,8(sp)
   10252:	40395913          	srai	s2,s2,0x3
   10256:	00090b63          	beqz	s2,1026c <__libc_init_array+0x34>
   1025a:	4f040413          	addi	s0,s0,1264
   1025e:	4481                	li	s1,0
   10260:	601c                	ld	a5,0(s0)
   10262:	0485                	addi	s1,s1,1
   10264:	0421                	addi	s0,s0,8
   10266:	9782                	jalr	a5
   10268:	fe991ce3          	bne	s2,s1,10260 <__libc_init_array+0x28>
   1026c:	6445                	lui	s0,0x11
   1026e:	6945                	lui	s2,0x11
   10270:	4f040793          	addi	a5,s0,1264 # 114f0 <__init_array_start>
   10274:	50090913          	addi	s2,s2,1280 # 11500 <__do_global_dtors_aux_fini_array_entry>
   10278:	40f90933          	sub	s2,s2,a5
   1027c:	40395913          	srai	s2,s2,0x3
   10280:	00090b63          	beqz	s2,10296 <__libc_init_array+0x5e>
   10284:	4f040413          	addi	s0,s0,1264
   10288:	4481                	li	s1,0
   1028a:	601c                	ld	a5,0(s0)
   1028c:	0485                	addi	s1,s1,1
   1028e:	0421                	addi	s0,s0,8
   10290:	9782                	jalr	a5
   10292:	fe991ce3          	bne	s2,s1,1028a <__libc_init_array+0x52>
   10296:	60e2                	ld	ra,24(sp)
   10298:	6442                	ld	s0,16(sp)
   1029a:	64a2                	ld	s1,8(sp)
   1029c:	6902                	ld	s2,0(sp)
   1029e:	6105                	addi	sp,sp,32
   102a0:	8082                	ret

00000000000102a2 <memset>:
   102a2:	433d                	li	t1,15
   102a4:	872a                	mv	a4,a0
   102a6:	02c37163          	bgeu	t1,a2,102c8 <memset+0x26>
   102aa:	00f77793          	andi	a5,a4,15
   102ae:	e3c1                	bnez	a5,1032e <memset+0x8c>
   102b0:	e1bd                	bnez	a1,10316 <memset+0x74>
   102b2:	ff067693          	andi	a3,a2,-16
   102b6:	8a3d                	andi	a2,a2,15
   102b8:	96ba                	add	a3,a3,a4
   102ba:	e30c                	sd	a1,0(a4)
   102bc:	e70c                	sd	a1,8(a4)
   102be:	0741                	addi	a4,a4,16
   102c0:	fed76de3          	bltu	a4,a3,102ba <memset+0x18>
   102c4:	e211                	bnez	a2,102c8 <memset+0x26>
   102c6:	8082                	ret
   102c8:	40c306b3          	sub	a3,t1,a2
   102cc:	068a                	slli	a3,a3,0x2
   102ce:	00000297          	auipc	t0,0x0
   102d2:	9696                	add	a3,a3,t0
   102d4:	00a68067          	jr	10(a3)
   102d8:	00b70723          	sb	a1,14(a4)
   102dc:	00b706a3          	sb	a1,13(a4)
   102e0:	00b70623          	sb	a1,12(a4)
   102e4:	00b705a3          	sb	a1,11(a4)
   102e8:	00b70523          	sb	a1,10(a4)
   102ec:	00b704a3          	sb	a1,9(a4)
   102f0:	00b70423          	sb	a1,8(a4)
   102f4:	00b703a3          	sb	a1,7(a4)
   102f8:	00b70323          	sb	a1,6(a4)
   102fc:	00b702a3          	sb	a1,5(a4)
   10300:	00b70223          	sb	a1,4(a4)
   10304:	00b701a3          	sb	a1,3(a4)
   10308:	00b70123          	sb	a1,2(a4)
   1030c:	00b700a3          	sb	a1,1(a4)
   10310:	00b70023          	sb	a1,0(a4)
   10314:	8082                	ret
   10316:	0ff5f593          	andi	a1,a1,255
   1031a:	00859693          	slli	a3,a1,0x8
   1031e:	8dd5                	or	a1,a1,a3
   10320:	01059693          	slli	a3,a1,0x10
   10324:	8dd5                	or	a1,a1,a3
   10326:	02059693          	slli	a3,a1,0x20
   1032a:	8dd5                	or	a1,a1,a3
   1032c:	b759                	j	102b2 <memset+0x10>
   1032e:	00279693          	slli	a3,a5,0x2
   10332:	00000297          	auipc	t0,0x0
   10336:	9696                	add	a3,a3,t0
   10338:	8286                	mv	t0,ra
   1033a:	fa2680e7          	jalr	-94(a3)
   1033e:	8096                	mv	ra,t0
   10340:	17c1                	addi	a5,a5,-16
   10342:	8f1d                	sub	a4,a4,a5
   10344:	963e                	add	a2,a2,a5
   10346:	f8c371e3          	bgeu	t1,a2,102c8 <memset+0x26>
   1034a:	b79d                	j	102b0 <memset+0xe>

000000000001034c <__call_exitprocs>:
   1034c:	715d                	addi	sp,sp,-80
   1034e:	f052                	sd	s4,32(sp)
   10350:	f481ba03          	ld	s4,-184(gp) # 11c50 <_global_impure_ptr>
   10354:	f84a                	sd	s2,48(sp)
   10356:	e486                	sd	ra,72(sp)
   10358:	1f8a3903          	ld	s2,504(s4)
   1035c:	e0a2                	sd	s0,64(sp)
   1035e:	fc26                	sd	s1,56(sp)
   10360:	f44e                	sd	s3,40(sp)
   10362:	ec56                	sd	s5,24(sp)
   10364:	e85a                	sd	s6,16(sp)
   10366:	e45e                	sd	s7,8(sp)
   10368:	e062                	sd	s8,0(sp)
   1036a:	02090863          	beqz	s2,1039a <__call_exitprocs+0x4e>
   1036e:	8b2a                	mv	s6,a0
   10370:	8bae                	mv	s7,a1
   10372:	4a85                	li	s5,1
   10374:	59fd                	li	s3,-1
   10376:	00892483          	lw	s1,8(s2)
   1037a:	fff4841b          	addiw	s0,s1,-1
   1037e:	00044e63          	bltz	s0,1039a <__call_exitprocs+0x4e>
   10382:	048e                	slli	s1,s1,0x3
   10384:	94ca                	add	s1,s1,s2
   10386:	020b8663          	beqz	s7,103b2 <__call_exitprocs+0x66>
   1038a:	2084b783          	ld	a5,520(s1)
   1038e:	03778263          	beq	a5,s7,103b2 <__call_exitprocs+0x66>
   10392:	347d                	addiw	s0,s0,-1
   10394:	14e1                	addi	s1,s1,-8
   10396:	ff3418e3          	bne	s0,s3,10386 <__call_exitprocs+0x3a>
   1039a:	60a6                	ld	ra,72(sp)
   1039c:	6406                	ld	s0,64(sp)
   1039e:	74e2                	ld	s1,56(sp)
   103a0:	7942                	ld	s2,48(sp)
   103a2:	79a2                	ld	s3,40(sp)
   103a4:	7a02                	ld	s4,32(sp)
   103a6:	6ae2                	ld	s5,24(sp)
   103a8:	6b42                	ld	s6,16(sp)
   103aa:	6ba2                	ld	s7,8(sp)
   103ac:	6c02                	ld	s8,0(sp)
   103ae:	6161                	addi	sp,sp,80
   103b0:	8082                	ret
   103b2:	00892783          	lw	a5,8(s2)
   103b6:	6498                	ld	a4,8(s1)
   103b8:	37fd                	addiw	a5,a5,-1
   103ba:	04878463          	beq	a5,s0,10402 <__call_exitprocs+0xb6>
   103be:	0004b423          	sd	zero,8(s1)
   103c2:	db61                	beqz	a4,10392 <__call_exitprocs+0x46>
   103c4:	31092783          	lw	a5,784(s2)
   103c8:	008a96bb          	sllw	a3,s5,s0
   103cc:	00892c03          	lw	s8,8(s2)
   103d0:	8ff5                	and	a5,a5,a3
   103d2:	2781                	sext.w	a5,a5
   103d4:	ef89                	bnez	a5,103ee <__call_exitprocs+0xa2>
   103d6:	9702                	jalr	a4
   103d8:	00892703          	lw	a4,8(s2)
   103dc:	1f8a3783          	ld	a5,504(s4)
   103e0:	01871463          	bne	a4,s8,103e8 <__call_exitprocs+0x9c>
   103e4:	fb2787e3          	beq	a5,s2,10392 <__call_exitprocs+0x46>
   103e8:	dbcd                	beqz	a5,1039a <__call_exitprocs+0x4e>
   103ea:	893e                	mv	s2,a5
   103ec:	b769                	j	10376 <__call_exitprocs+0x2a>
   103ee:	31492783          	lw	a5,788(s2)
   103f2:	1084b583          	ld	a1,264(s1)
   103f6:	8ff5                	and	a5,a5,a3
   103f8:	2781                	sext.w	a5,a5
   103fa:	e799                	bnez	a5,10408 <__call_exitprocs+0xbc>
   103fc:	855a                	mv	a0,s6
   103fe:	9702                	jalr	a4
   10400:	bfe1                	j	103d8 <__call_exitprocs+0x8c>
   10402:	00892423          	sw	s0,8(s2)
   10406:	bf75                	j	103c2 <__call_exitprocs+0x76>
   10408:	852e                	mv	a0,a1
   1040a:	9702                	jalr	a4
   1040c:	b7f1                	j	103d8 <__call_exitprocs+0x8c>

000000000001040e <__libc_fini_array>:
   1040e:	1101                	addi	sp,sp,-32
   10410:	e822                	sd	s0,16(sp)
   10412:	67c5                	lui	a5,0x11
   10414:	6445                	lui	s0,0x11
   10416:	50040413          	addi	s0,s0,1280 # 11500 <__do_global_dtors_aux_fini_array_entry>
   1041a:	50878793          	addi	a5,a5,1288 # 11508 <impure_data>
   1041e:	8f81                	sub	a5,a5,s0
   10420:	e426                	sd	s1,8(sp)
   10422:	ec06                	sd	ra,24(sp)
   10424:	4037d493          	srai	s1,a5,0x3
   10428:	c881                	beqz	s1,10438 <__libc_fini_array+0x2a>
   1042a:	17e1                	addi	a5,a5,-8
   1042c:	943e                	add	s0,s0,a5
   1042e:	601c                	ld	a5,0(s0)
   10430:	14fd                	addi	s1,s1,-1
   10432:	1461                	addi	s0,s0,-8
   10434:	9782                	jalr	a5
   10436:	fce5                	bnez	s1,1042e <__libc_fini_array+0x20>
   10438:	60e2                	ld	ra,24(sp)
   1043a:	6442                	ld	s0,16(sp)
   1043c:	64a2                	ld	s1,8(sp)
   1043e:	6105                	addi	sp,sp,32
   10440:	8082                	ret

0000000000010442 <atexit>:
   10442:	85aa                	mv	a1,a0
   10444:	4681                	li	a3,0
   10446:	4601                	li	a2,0
   10448:	4501                	li	a0,0
   1044a:	a009                	j	1044c <__register_exitproc>

000000000001044c <__register_exitproc>:
   1044c:	f481b703          	ld	a4,-184(gp) # 11c50 <_global_impure_ptr>
   10450:	1f873783          	ld	a5,504(a4)
   10454:	c3b1                	beqz	a5,10498 <__register_exitproc+0x4c>
   10456:	4798                	lw	a4,8(a5)
   10458:	487d                	li	a6,31
   1045a:	06e84263          	blt	a6,a4,104be <__register_exitproc+0x72>
   1045e:	c505                	beqz	a0,10486 <__register_exitproc+0x3a>
   10460:	00371813          	slli	a6,a4,0x3
   10464:	983e                	add	a6,a6,a5
   10466:	10c83823          	sd	a2,272(a6)
   1046a:	3107a883          	lw	a7,784(a5)
   1046e:	4605                	li	a2,1
   10470:	00e6163b          	sllw	a2,a2,a4
   10474:	00c8e8b3          	or	a7,a7,a2
   10478:	3117a823          	sw	a7,784(a5)
   1047c:	20d83823          	sd	a3,528(a6)
   10480:	4689                	li	a3,2
   10482:	02d50063          	beq	a0,a3,104a2 <__register_exitproc+0x56>
   10486:	00270693          	addi	a3,a4,2
   1048a:	068e                	slli	a3,a3,0x3
   1048c:	2705                	addiw	a4,a4,1
   1048e:	c798                	sw	a4,8(a5)
   10490:	97b6                	add	a5,a5,a3
   10492:	e38c                	sd	a1,0(a5)
   10494:	4501                	li	a0,0
   10496:	8082                	ret
   10498:	20070793          	addi	a5,a4,512
   1049c:	1ef73c23          	sd	a5,504(a4)
   104a0:	bf5d                	j	10456 <__register_exitproc+0xa>
   104a2:	3147a683          	lw	a3,788(a5)
   104a6:	4501                	li	a0,0
   104a8:	8e55                	or	a2,a2,a3
   104aa:	00270693          	addi	a3,a4,2
   104ae:	068e                	slli	a3,a3,0x3
   104b0:	2705                	addiw	a4,a4,1
   104b2:	30c7aa23          	sw	a2,788(a5)
   104b6:	c798                	sw	a4,8(a5)
   104b8:	97b6                	add	a5,a5,a3
   104ba:	e38c                	sd	a1,0(a5)
   104bc:	8082                	ret
   104be:	557d                	li	a0,-1
   104c0:	8082                	ret

00000000000104c2 <_exit>:
   104c2:	05d00893          	li	a7,93
   104c6:	00000073          	ecall
   104ca:	00054363          	bltz	a0,104d0 <_exit+0xe>
   104ce:	a001                	j	104ce <_exit+0xc>
   104d0:	1141                	addi	sp,sp,-16
   104d2:	e022                	sd	s0,0(sp)
   104d4:	842a                	mv	s0,a0
   104d6:	e406                	sd	ra,8(sp)
   104d8:	4080043b          	negw	s0,s0
   104dc:	008000ef          	jal	ra,104e4 <__errno>
   104e0:	c100                	sw	s0,0(a0)
   104e2:	a001                	j	104e2 <_exit+0x20>

00000000000104e4 <__errno>:
   104e4:	f581b503          	ld	a0,-168(gp) # 11c60 <_impure_ptr>
   104e8:	8082                	ret
