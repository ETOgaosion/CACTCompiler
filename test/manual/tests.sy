int add_int(int a, int b)
{
    int c;
    c = a + b;
    return c;
}
int main()
{
    int i = 0;
    int j = 0;
    int b;
    while(i < 10 && j < 1){
        if(i < 5){
            b = add_int(i,j);
        } else{
            b = 0;
        }
        i = i + 2;
        j = j + 1;
    }
    return 0;
}