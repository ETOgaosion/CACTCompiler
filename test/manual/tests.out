bbl loader
1
8
